src/librustc_codegen_llvm/debuginfo/metadata.rs

```diff
@@ -1370,7 +1370,7 @@ impl EnumMemberDescriptionFactory<'ll, 'tcx> {
                                 .wrapping_sub(niche_variants.start().as_u32() as u128)
                                 .wrapping_add(niche_start);
                             let value = truncate(value, niche.value.size(cx));
-                            Some(value as u64)
+                            Some(value as u128)
```

Comment:
You should be able to remove the cast here (since `value: u128`) to make it mode obvious this is all lossless.

Reply:
You can even write `Some(truncate(...))` since the intermediary variable only existed for the assert.